src/test/clojure/core2/temporal/kd_tree_test.clj

```diff
@@ -312,13 +353,535 @@
                            :app-time-end #inst "1998-01-15T00:00:00.000-00:00",
                            :sys-time-start #inst "1998-01-28T00:00:00.000-00:00",
                            :sys-time-end #inst "9999-12-31T23:59:59.999-00:00"}]
-                         (temporal-rows kd-tree row-id->row))
+                         (temporal-rows kd-tree row-id->row)))
+                (t/is (= #{}
+                         @!current-row-ids))
+
 
-                      (t/testing "rebuilding tree results in tree with same points"
-                        (let [points (mapv vec (kd/kd-tree->seq kd-tree))]
-                          (with-open [rebuilt-tree (kd/build-node-kd-tree allocator (shuffle points))]
-                            (t/is (= (sort points)
-                                     (sort (mapv vec (kd/kd-tree->seq rebuilt-tree)))))))))))))))))
+                (t/testing "rebuilding tree results in tree with same points"
+                  (let [points (mapv vec (kd/kd-tree->seq kd-tree))]
+                    (with-open [rebuilt-tree (kd/build-node-kd-tree allocator (shuffle points))]
+                      (t/is (= (sort points)
+                               (sort (mapv vec (kd/kd-tree->seq rebuilt-tree))))))))))))))))
+
+(defn current-rows-for [sys-time inserts]
```

Comment:
good set of tests :slightly_smiling_face: could/should some of these live in `core2.temporal-test` instead, given they're testing fns in there rather than the kd-tree itself?

Reply:
I'd also be tempted to use the test-util `*allocator*` fixture, rather than you faffing about with `with-open` etc every time

Reply:
Yes, they probably all belong in current_row_ids_test.clj